### ISSUE TITLE
How to open a shared, in-memory database with more recent sqlite3 versions?

### ISSUE THREAD

**Issue by KasperJanssens** [jgallagher/rusqlite#616]
Hello, I found issue #135 on your issue list and I was trying the proposed work around with a later version of sqlite3, totally ignoring the fact that you mentioned this would not work. Turns out that this fails, no errors but the database has clearly not been shared as I get a message about tables not existing when I use my second connection. So it seems like it defaulted to non-shared.
Is there a way to get this to work with later versions or not really?

**Comment by gwenn** [comment]
It depends only on the method/url used to open the database and not on sqlite3 version.
See https://github.com/jgallagher/rusqlite/blob/master/src/unlock_notify.rs#L114 for an example that works.

**Comment by gwenn** [comment]
https://sqlite.org/inmemorydb.html
> In-memory databases are allowed to use shared cache if they are opened using a URI filename.

**Comment by KasperJanssens** [comment]
I had tried exactly that in my application and it did not work. I wrote a small test based upon your example and that does work. So, I have likely made a mistake in my application. If it is pertinent I will come back and share it. If it is plain embarrassing I'll just close the issue. Will get back to you later, time's up for today. Thanks for the quick response.

**Comment by KasperJanssens** [comment]
Okay, I get it, I opened two connections but by the time I opened the second one the first one had already gone out of scope due to a move operation and as such the content was cleaned up. Making sure that I open both connections before I move one of them away seems to be the answer. So something in between an interesting share and something I should have known. The context was a connection for the application and a connection for test asserting, and the asserting connection was opened after the application had run, by the way. Thanks for the help.
